src/compiler/checker.ts

```diff
@@ -11357,7 +11357,20 @@ module ts {
 
         function checkGrammarFunctionLikeDeclaration(node: FunctionLikeDeclaration): boolean {
             // Prevent cascading error by short-circuit
-            return checkGrammarModifiers(node) || checkGrammarTypeParameterList(node, node.typeParameters) || checkGrammarParameterList(node.parameters);
+            return checkGrammarModifiers(node) || checkGrammarTypeParameterList(node, node.typeParameters) || checkGrammarParameterList(node.parameters) || checkGrammarArrowFunction(node);
+        }
+
+        function checkGrammarArrowFunction(node: FunctionLikeDeclaration): boolean {
+            if (node.kind === SyntaxKind.ArrowFunction) {
+                var arrowFunction = <ArrowFunction>node;
+                var sourceFile = getSourceFileOfNode(node);

```

Comment:
I actually recall one of the other functions needing the sourceFile as well. Could we grab the sourceFile once in `checkGrammarFunctionLikeDeclaration` and pass it in as a parameter to these checkers?